pkg/webhooks/policymutation.go

```diff
@@ -302,10 +303,10 @@ func generateRuleForControllers(rule kyverno.Rule, controllers string, log logr.
 	}
 
 	// overwrite Kinds by pod controllers defined in the annotation
-	controllerRule.MatchResources.Kinds = strings.Split(controllers, ",")
+	controllerRule.MatchResources.Kinds = replacePodWithPodControllers(controllerRule.MatchResources.Kinds, strings.Split(controllers, ","))
```

Comment:
Why do we need to pass in `kinds` in matched resourceDescription? The auto-gen only generates the rule for defined controllers.

Reply:
Will revert this change, this is a poor solution to a problem i thought i had.

Reply:
requested changes have been made